sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs

```diff
@@ -611,6 +615,20 @@ public partial class ShareFileCopyInfo
         public Azure.ETag ETag { get { throw null; } }
         public System.DateTimeOffset LastModified { get { throw null; } }
     }
+    public partial class ShareFileCopyOptions
+    {
+        public ShareFileCopyOptions() { }
+        public Azure.Storage.Files.Shares.Models.ShareFileRequestConditions Conditions { get { throw null; } set { } }
+        public bool? CopySourceFileAttributes { get { throw null; } set { } }
+        public bool? CopySourceFileCreatedOn { get { throw null; } set { } }
+        public bool? CopySourceFileLastWrittenOn { get { throw null; } set { } }
```

Comment:
This is a tricky problem.  How important do we think setting `source` will be for these properties?  If it's very important, maybe we stick with very visible properties.

Otherwise I worry that these properties both take up a bit of space in the Options bag and won't be as obvious to customers when they're setting SmbProperties.  What if we did something like:

```C#
public partial class ShareFileCopyOptions
{
     // ...
     public FileSmbProperties SmbProperties { get; set; }
     public CopyableFileSmbProperties SmbPropertiesToCopy { get; set; }
}

[Flags]
public enum CopyableFileSmbProperties
{
    None = 0,
    CreatedOn = 1,
    LastWrittenOn = 2,
    FileAttributes = 4,
}
```

Then they'll appear next to each other in intellisense.

Reply:
Have any thoughts on the best way to merge this into the GA'ed API @KrzysztofCwalina?

Reply:
> This is a tricky problem. How important do we think setting source will be for these properties?

I would say not very important.  We currently don't have this feature in the SDKs, and we haven't gotten any GitHub issues about it.